src/t8_vtk/t8_vtk_reader.cxx

```diff
@@ -426,6 +426,9 @@ t8_vtkGrid_to_cmesh (vtkSmartPointer < vtkDataSet >vtkGrid,
 vtkSmartPointer < vtkPointSet >
 t8_vtkGrid_to_vtkPointSet (vtkSmartPointer < vtkDataSet >vtkGrid)
 {
+  if (vtkGrid == NULL) {
+    return NULL;
+  }
```

Comment:
Shouldn`t the function return a `vtkSmartPointer` pointing to `NULL`?

Reply:
That would call the destructor of the vtkSmartPointer, so we can return NULL directly.